### PR TITLE
openssl_1_1: use less

### PR DESCRIPTION
**Pull request by ajs124**
###### Description of changes
This should hopefully make it so updating `openssl_1_1` is not a mass rebuild on linux and and stdenv rebuild on darwin anymore.

Sadly I don't think I documented the specific python packages that motivated me to commit 8dfc998207dca2e2928cd1f204e4f37444aa40a6.

Edit: I built `nixosTests.openldap` on x86_64-linux. See comments for more testing.

###### Things done
- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/cyrus-sasl/default.nix [hunk 1]
```diff
@@ -19,6 +19,8 @@ stdenv.mkDerivation rec {
   patches = [
     # Fix cross-compilation
     ./cyrus-sasl-ac-try-run-fix.patch
+    # make compatible with openssl3. can probably be dropped with any release after 2.1.28
+    ./openssl3.patch
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/cyrus-sasl/default.nix:23]: I think we should be able to fetchpatch here.

File: pkgs/development/libraries/cyrus-sasl/openssl3.patch [hunk 2]
```diff
@@ -0,0 +1,196 @@
+From 455417ad5d7da87d22590942a433939bdff986ca Mon Sep 17 00:00:00 2001
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/cyrus-sasl/openssl3.patch:1]: Upstream commit seems to be https://github.com/cyrusimap/cyrus-sasl/commit/887dbc0435056ec58ee48c4d803f110ade1e4c39

**Review by SuperSandro2000 — COMMENTED** [review]
Not sure when the next python-updates run happens but we could push the cpython change into there to find regressions easy.

File: pkgs/development/libraries/cyrus-sasl/openssl3.patch (hunk 2 above)
Comment (ajs124) [reply pkgs/development/libraries/cyrus-sasl/openssl3.patch:1]: It's three commits https://github.com/cyrusimap/cyrus-sasl/commit/887dbc0435056ec58ee48c4d803f110ade1e4c39, https://github.com/cyrusimap/cyrus-sasl/commit/dfaa62392e7caecc6ecf0097b4d73738ec4fc0a8 and https://github.com/cyrusimap/cyrus-sasl/commit/c2bd3afbca57f176d8c650670ce371444bb7fcc0

**Comment by ajs124** [comment]
> Not sure when the next python-updates run happens but we could push the cpython change into there to find regressions easy.

cc @mweinelt

**Comment by mweinelt** [comment]
Ideally a minor update run that goes into the next staging-next cycle, but then we'd ideally need to start today.

**Comment by SuperSandro2000** [comment]
Verified the python change exemplary with ``nix build .#python3.pkgs.uamqp .#python3.pkgs.cryptography .#python3.pkgs.m2crypto`` which all succeeded.

File: pkgs/development/libraries/cyrus-sasl/openssl3.patch (hunk 2 above)
Comment (Atemu) [reply pkgs/development/libraries/cyrus-sasl/openssl3.patch:1]: https://patch-diff.githubusercontent.com/raw/cyrusimap/cyrus-sasl/pull/653.patch

You can do the same on a range of commits if you want to pin it.

**Review by SuperSandro2000 — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**